[FIX] mrp_subcontracting: Put source locations, and correct destinati…

alfredoavanzosc opened the pull request:
…on, where products are added to an operation.

Se ha arreglado un BUG reportado por Ana. El caso es que cuando se estaban añadiendo productos a una operación de subcontratación, se estaban añadiendo mal las ubicaciones al nuevo movimiento que se estaba dando de alta.
Este BUG lo reportó Ana en el issue https://github.com/odoomrp/odoomrp-wip/issues/918

alfredoavanzosc committed:
[FIX] mrp_subcontracting: Put source locations, and correct destination, where products are added to an operation.

pedrobaeza commented:
No se debería añadir la dependencia a ese módulo en el de subcontratación, ya que son independientes y aquellos que tengan instalado el módulo les cascará en el momento en el que actualicen. Hay dos opciones:
- Un nuevo glue module para unir ambos. Un mal necesario para respetar la herencia.
- Tener unos campos a nivel de WO que sean de donde se cogen las ubicaciones en caso de añadir manualmente. Esos campos se escribirían con el valor adecuado en el caso de la subcontratación. Esta opción evita el módulo extra. Para ello, habría que poner los campos en el _mrp_operations_extension_, que el _mrp_subcontracting_ sobreescriba el valor, y que el add_middle_stuff utilice esa campo.

anajuaristi commented:
Ya vimos el tema de la dependencia y fui yo la que dijo a Alfredo de
hacerlo asi para evitar otro glue module, entendiendo que todos los
clientes se han instalado el middle stuff pero no todos tienen
subcontracting
Me gusta la segunda opcion pero si lo hacemos no solo deberia ser para el
caso de subcontratacion sino en general para todos los casos.
Dando a la misma idea una vuelta de tuerca mas, se me ocurre que por
estandar en v8 las ubicaciones origen y destino estan en el tipo de
operacion. Este campo lo añade a la ruta el modulo de subcontracting pero
si lo añadiese el operations extension ya no nos haria falta el glue module
ni meter los dos campos a pelo en la operacion.
Ademas permitiria que si ponemos un tipo de picking en cualquier operacion
de la ruta, si alguien lo necesita los movimientos se podrian incluir
facilmente en un albaran aunque la operacion no sea externa.
Tambien nos predispone el sistema para hacer dissasembly con facilidad (me
lo han pedido hoy). De esta forma, si sacan piezas de algo en una operacion
y las ubicaciones origen y destino en esa operacion estan al reves no
habroa que hacer nada mas. El requerimiento quedaria cubierto.
Me gusta este enfoque. Como lo veis entonces?
 El 29/07/2015 19:10, "Pedro M. Baeza" notifications@github.com escribió:

> No se debería añadir la dependencia a ese módulo en el de subcontratación,
> ya que son independientes y aquellos que tengan instalado el módulo les
> cascará en el momento en el que actualicen. Hay dos opciones:
> - Un nuevo glue module para unir ambos. Un mal necesario para respetar
>   la herencia.
> - Tener unos campos a nivel de WO que sean de donde se cogen las
>   ubicaciones en caso de añadir manualmente. Esos campos se escribirían con
>   el valor adecuado en el caso de la subcontratación. Esta opción evita el
>   módulo extra. Para ello, habría que poner los campos en el
>   _mrp_operations_extension_, que el _mrp_subcontracting_ sobreescriba
>   el valor, y que el add_middle_stuff utilice esa campo.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/odoomrp/odoomrp-wip/pull/925#issuecomment-126021410.

pedrobaeza commented:
Adelante entonces

alfredoavanzosc committed:
[IMP] mrp_subcontrating: Remove the creation of the field "picking_type_id" in mrp.routing.workcenter.
[IMP] mrp_operations_extension: Creation of the field "picking_type_id" in mrp.routing.workcenter.

alfredoavanzosc commented:
He quitado la creación del campo "picking_type_id" de "mrp_subcontracting", y lo he creado en "mrp_operations_extension", además este último módulo lleva la modificación solicitada en CLM0311-Añadir concepto en reclamaciones: En el campo "referencia" de reclamaciones, añadir al listado el modelo "workcenter".

pedrobaeza commented on mrp_subcontracting/models/mrp_production.py:
```diff
@@ -67,6 +53,23 @@ def action_confirm(self):
+            work_order = self.env.context.get('work_order')
+            if work_order.routing_wc_line.external:
+                picking_type = work_order.routing_wc_line.picking_type_id
+                vals = {'location_id': picking_type.default_location_src_id.id,
```
Puesto que el campo no es obligatorio, habría que proveer de un valor alternativo en caso de que picking_type sea False

pedrobaeza replied:
O lo coge ya de por si?

anajuaristi replied:
Si el picking type es false, debería de coger las ubicaciones de forma standar, o sea, las que están en la cabecera de la OF. 
Por defecto lo hace así o sea que entiendo que sí que lo cogería aunque no he visto el código.

pedrobaeza replied:
No, pero hay que comprobar `if picking_type:` en este código, porque si no, va a cambiar ese valor por defecto a False.

anajuaristi replied:
Ahora que lo veo, solo entra al IF, si la operación es externa y en ese caso el picking type es obligatorio por lo que lo cogería bien pero quizás no esté de más preguntar también el if picking_type

pedrobaeza replied:
No viene mal, por si programáticamente (o al migrar) se ha hecho algo mal, aunque por interfaz el dato esté bien.

alfredoavanzosc committed:
[FIX] mrp_subcontracting: Improve programming.

alfredoavanzosc commented:
Realizada la mejora solicitada por Pedro, es decir, ampliar la condición if (work_order.routing_wc_line.external and work_order.routing_wc_line.picking_type_id):

pedrobaeza commented:
Gracias! :+1:
